_aggregations/pipeline/bucket-script.md

```diff
@@ -0,0 +1,159 @@
+---
+layout: default
+title: Bucket script
+parent: Pipeline aggregations
+nav_order: 20
+redirect_from:
+  - /query-dsl/aggregations/pipeline-agg#bucket_script-bucket_selector/
+  - /query-dsl/aggregations/pipeline/bucket-selector/
+---
+
+# Bucket script aggregations
+
+The `bucket_script` aggregation is a parent pipeline aggregation that executes a script to perform per-bucket numeric computations across a set of buckets. Use the `bucket_script` aggregation to perform custom numeric computations on multiple metrics in a bucketed aggregation. For example, you can:
+
+- Calculate derived and composite metrics
+- Apply conditional logic using if/else statements
+- Compute business-specific KPIs such as custom scoring metrics
+
+## Parameters
+
+The `bucket_script` aggregation takes the following parameters:
+
+| Parameter             | Required/Optional | Data type       | Description |
+| :--                   | :--               |  :--            | :--         |
+| `buckets_path`        | Required          | Object          | A map of variable names to bucketed metrics that identify the metrics to be used in the script. The metrics must be numeric. See [Script variables](#script-variables). |
+| `script`              | Required          | String or Object | The script to execute. Can be an in-line script, stored script, or script file. The script has access to the variable names defined in the `buckets_path` parameter. Must return a numeric value. |
+| `gap_policy`          | Optional          | String          | The policy to apply to missing data. Valid values are `skip`, `insert_zeros`, and `keep_values`. Default is `skip`. |
+| `format`              | Optional          | String          | A [DecimalFormat](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/text/DecimalFormat.html) formatting string. Returns the formatted output in the aggregation's `value_as_string` property. |
+
+## Script variables
+
+The `buckets_path` parameter maps script variable names to metrics from parent aggregations. These variables can then be used in the script. For example, the following `buckets_path` maps the `sales_sum` metric to the `total_sales` script variable and the `item_count` metric to the `item_count` script variable:
```

Comment:
```suggestion
The `buckets_path` parameter maps script variable names to metrics from parent aggregations. These variables can then be used in the script. 

For the `bucket_script` and `bucket_selector` aggregations, the `buckets_path` parameter is an object rather than a string because it must refer to multiple bucket metrics. See the [Pipeline aggregations]({{site.url}}{{site.baseurl}}/aggregations/pipeline/index#buckets-path-property) page for a description of the string version of `buckets_path`.
{: .note}

The following `buckets_path` maps the `sales_sum` metric to the `total_sales` script variable and the `item_count` metric to the `item_count` script variable:
```

Reply:
Do I still need to explicitly specify even when the mapping is identical, like for `item_count`?

Reply:
My understanding is that you still need to explicitly specify the variable mapping in `buckets_path`, even if the variable name is identical to the aggregation name (e.g., `item_count`).

The `bucket_script` aggregation does not implicitly inherit variable names from existing aggregations or field mappings. It only knows what you explicitly pass in through `buckets_path`.